xncml/parser.py

```diff
@@ -0,0 +1,584 @@
+    ref : xr.Dataset
+      Reference dataset used to copy content into `target`.
+    obj : Netcdf
+       <netcdf> object description.
+    ncml : Path
+      Path to NcML document, sometimes required to follow relative links.
+
+    Returns
+    -------
+    xr.Dataset
+      Dataset holding variables and attributes defined in <netcdf> element.
+    """
+    # We need to start parsing <aggregation>, since it contains required information for later <variable> elements.
+    for item in filter_by_class(obj.choice, Aggregation):
+        target = read_aggregation(target, item, ncml)
+
+    target = read_group(target, ref, obj)
+
+    return target
+
+
+def read_aggregation(target: xr.Dataset, obj: Aggregation, ncml: Path) -> xr.Dataset:
+    """
+    Return merged or concatenated content of <aggregation> element.
+
+    Parameters
+    ----------
+    target : xr.Dataset
+      Target dataset to be updated with <netcdf>'s content.
+    obj : Aggregation
+       <aggregation> object description.
+    ncml : Path
+      Path to NcML document, sometimes required to follow relative links.
+
+    Returns
+    -------
+    xr.Dataset
+      Dataset holding variables and attributes defined in <aggregation> element.
+    """
+    from xarray.coding.times import CFDatetimeCoder
+
+    # Names of variables to be aggregated. All variables if undefined.
+    names = [v.name for v in filter_by_class(obj.choice, Aggregation.VariableAgg)]
+
+    # Create list of items to aggregate.
+    items = []
+    for item in obj.netcdf:
+        # Open dataset defined in <netcdf>'s `location` attribute
+        ref = read_ds(item, ncml) or xr.Dataset()
+
+        # Handle <explicit/> tag.
+        tar = xr.Dataset()
+        if item.read_metadata is not None:
+            tar = ref
+        elif item.explicit is not None:
+            pass
+        else:  # Default
+            tar = ref
+
+        # Handle <variable>, <attribute> and <remove> elements
+        tar = read_group(tar, ref, item)
+
+        # Select variables
+        if names:
+            tar = tar[names]
+
+        # Handle coordinate values
+        if item.coord_value is not None:
+            dtypes = [i[obj.dim_name].dtype.type for i in [tar, target, ref] if obj.dim_name in i]
+            coords = read_coord_value(item, obj, dtypes=dtypes)
+            tar = tar.assign_coords({obj.dim_name: coords})
+        items.append(tar)
+
+    # Handle <scan> element
+    for item in obj.scan:
+        items.extend(read_scan(item, ncml))
+
+    # Need to decode time variable
+    if obj.time_units_change:
+        for i, ds in enumerate(items):
+            t = xr.as_variable(ds[obj.dim_name], obj.dim_name)  # Maybe not the same name...
+            encoded = CFDatetimeCoder(use_cftime=True).decode(t, name=t.name)
+            items[i] = ds.assign_coords({obj.dim_name: encoded})
+
+    # Translate different types of aggregation into xarray instructions.
+    match obj.type:
```

Comment:
As discussed earlier, it's cool to use the `match` structure, but it is new in python 3.10  and the current numpy/xarray is still accepting python 3.8.
According to NEP 29, Python 3.10+ is scheduled for Apr 5, 2024 ;).

Reply:
Will fix.